src/Entities/Player.cpp

```diff
@@ -146,6 +146,22 @@ cPlayer::cPlayer(cClientHandlePtr a_Client, const AString & a_PlayerName) :
 
 
 
+void cPlayer::InitializePlayer()
+{
+	ASSERT(GetWorld() != nullptr);
+	ASSERT(GetParentChunk() == nullptr);
+	GetWorld()->AddPlayer(this);
+
+	cPluginManager::Get()->CallHookSpawnedEntity(*GetWorld(), *this);
+
+	// Spawn the entity on the clients:
+	GetWorld()->BroadcastSpawnEntity(*this);
```

Comment:
On a slightly related note, I think the teleport packets could do some tweaking. Sometimes I get nether fog in overworlds or vice versa. Someone reported skeleton horses, I suspect this could be wrong spawn packets or something.